pkgs/btcpayserver/default.nix

```diff
@@ -0,0 +1,79 @@
+{ stdenv, fetchFromGitHub, fetchurl, nbxplorer, dotnetPackages, dotnetCorePackages, lib }:
+
+let
+
+  deps = import ./deps.nix {inherit fetchurl;};
+  dotnet-sdk = dotnetCorePackages.sdk_3_1;
+
+in
+
+stdenv.mkDerivation rec {
+  name = "btcpayserver";
+  version = "1.0.5.4";
+
+  src = fetchFromGitHub {
+    owner = "btcpayserver";
+    repo = "btcpayserver";
+    rev = "v${version}";
+    sha256 = "1h4567di6sfr6z734b7a7qh0hv5ywq14hlbimxcfxn9hri2g3zbv";
+  };
+  buildInputs = [dotnet-sdk dotnetPackages.Nuget];
+
+  buildPhase = ''
+    mkdir home
+    export HOME=$PWD/home
+
+    export DOTNET_CLI_TELEMETRY_OPTOUT=1
+    export DOTNET_SKIP_FIRST_EXPERIENCE=1
+
+    for package in ${toString deps}; do
+      nuget add $package -Source nixos
+    done
+
+    dotnet restore --source nixos BTCPayServer.Client/BTCPayServer.Client.csproj
+    dotnet build --no-restore -c Release BTCPayServer.Client/BTCPayServer.Client.csproj
+
+    dotnet restore --source nixos BTCPayServer.Data/BTCPayServer.Data.csproj
+    dotnet build --no-restore -c Release BTCPayServer.Data/BTCPayServer.Data.csproj
+
+    dotnet restore --source nixos BTCPayServer.Common/BTCPayServer.Common.csproj
+    dotnet build --no-restore -c Release BTCPayServer.Common/BTCPayServer.Common.csproj
+
+    dotnet restore --source nixos BTCPayServer.Rating/BTCPayServer.Rating.csproj
+    dotnet build --no-restore -c Release BTCPayServer.Rating/BTCPayServer.Rating.csproj
+
+    dotnet restore --source nixos BTCPayServer/BTCPayServer.csproj
+    dotnet build --no-restore -c Release BTCPayServer/BTCPayServer.csproj
+  '';
+
+  installPhase = ''
+      mkdir -p $out/bin/
+      echo -e '#!/usr/bin/env bash \n#launch BTCpayserver \nscriptdir=`dirname $(readlink $(which btcpayserver))` \ndotnet exec --additionalprobingpath $scriptdir/../home/.nuget/packages/ $scriptdir/../BTCPayServer/bin/Release/netcoreapp3.1/BTCPayServer.dll $@' > $out/bin/run.sh
+      chmod +x $out/bin/run.sh
+      cp -r ./* $out
+      ln -s $out/bin/run.sh $out/bin/btcpayserver
+  '';
+
+  dontStrip = true;
+
+  meta = with lib; {
+    description = "BTCPay Server is a self-hosted, open-source cryptocurrency payment processor. It's secure, private, censorship-resistant and free";
+    longDescription = ''
+      BTCPay Server is a free and open-source cryptocurrency payment processor which allows you to receive payments in Bitcoin and altcoins directly, with no fees, transaction cost or a middleman.
+
```

Comment:
Is it normal to include newlines between longDescription lines?

Reply:
I don't think there are any standards for longDescription. Will remove them